### PR TITLE
feat(core): add optional chips filters to table

### PR DESCRIPTION
**Pull request by grankovsky**

### CODE REVIEW

File: packages/components/table/components/xm-quick-filters-control.component.ts [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+        this.destroy$.complete();
+    }
+
+    public ngOnInit(): void {
+        this.initForm();
+        this.subscribeFormChanges();
+    }
+
+    public ngOnChanges(changes: SimpleChanges): void {
+        ///debugger on change
```
Comment (denisogolubev) [inline packages/components/table/components/xm-quick-filters-control.component.ts:95]: comment

File: packages/components/table/components/xm-quick-filters-control.component.ts [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+                if (!_.isEqual(this.value, value)) {
+                    this.emitValue(value);
+                }
+
+                this.validStatusChange.emit(this.formGroup.valid);
+            });
+        }
+    }
+
+    private emitValue(value): void {
```
Comment (denisogolubev) [inline packages/components/table/components/xm-quick-filters-control.component.ts:176]: ESLint: Argument `value` should be typed. Run `lint` to check all the issues you have in code.

File: packages/components/table/components/xm-table-filter-inline.component.ts [hunk 3]
```diff
@@ -101,10 +96,17 @@ export class XmTableFilterInlineComponent implements OnInit, OnDestroy {
     private request$: Subject<FiltersControlValue> = new Subject<FiltersControlValue>();
 
     private tableFilterController: XmTableFilterController = inject(XmTableFilterController);
+    public isFilterVisible:boolean = false;
 
     public ngOnInit(): void {
         this.filterExpand = !this.config.isOnlyExpand;
         this.value = this.tableFilterController.get();
+        this.tableFilterController.filterVisibility$
```
Comment (denisogolubev) [inline packages/components/table/components/xm-table-filter-inline.component.ts:104]: Move this code to separate method. Don't expand code in lifecycle hooks.

File: packages/components/table/components/xm-table-quick-filter-controls.component.ts [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+import {
+    FiltersControlValue,
+    XmTableFilterButtonDialogControlComponent
+} from './xm-table-filter-button-dialog-control.component';
+import { FormLayoutItem } from '@xm-ngx/components/form-layout';
+import { JsonPipe } from '@angular/common';
+import { XmTableQuickFilterControl } from '../components/xm-quick-filters-control.component';
+import { MatChipsModule } from '@angular/material/chips';
+
+export interface XmTableFiltersControlRequestConfig {
```
Comment (denisogolubev) [inline packages/components/table/components/xm-table-quick-filter-controls.component.ts:11]: should be in `model` file

File: packages/components/table/components/xm-table-quick-filter-controls.component.ts [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+    @Output() public filtersChanged: EventEmitter<boolean> = new EventEmitter<boolean>();
+
+    @Input() public disabled: boolean;
+    @Input() public loading: boolean;
+    @Input() public options: any;
+
+    public value: FiltersControlValue;
+    public valid: boolean = true;
+
+    public OnValueChange(value: FiltersControlValue): void {
```
Comment (denisogolubev) [inline packages/components/table/components/xm-table-quick-filter-controls.component.ts:56]: use camelCase for the naming

File: packages/components/table/components/xm-table-quick-filter-controls.component.ts [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+    }
+
+    public ngOnChanges(changes: SimpleChanges): void {
+        if (changes.request) {
+            this.value = this.request;
+        }
+    }
+
+    public ngOnInit(): void {
+        this.options;
```
Comment (denisogolubev) [inline packages/components/table/components/xm-table-quick-filter-controls.component.ts:68]: What this mean?

File: packages/components/table/components/xm-table-quick-filter-inline.component.ts [hunk 7]
```diff
@@ -0,0 +1,170 @@
...
+    private request$: Subject<FiltersControlValue> = new Subject<FiltersControlValue>();
+
+    private tableFilterController: XmTableFilterController = inject(XmTableFilterController);
+    public isFilterVisible: boolean = false;
+    public filterSubscription: Subscription;
+
+    public ngOnInit(): void {
+        this.filterExpand = !this.config.isOnlyExpand;
+        this.value = this.tableFilterController.get();
+        this.filterSubscription = this.tableFilterController.filterVisibility$.subscribe(
```
Comment (denisogolubev) [inline packages/components/table/components/xm-table-quick-filter-inline.component.ts:113]: Subscription handling?

File: packages/components/table/components/xm-table-quick-filter-inline.component.ts [hunk 8]
```diff
@@ -0,0 +1,170 @@
...
+            .pipe(
+                takeUntilOnDestroy(this)
+            )
+            .subscribe((value: FiltersControlValue) => {
+                this.value = _.merge({}, this.cacheFilters, value);
+            });
+    }
+
+    private submitOnChangeFilter(): void {
+        this.request$.asObservable().pipe(
```
Comment (denisogolubev) [inline packages/components/table/components/xm-table-quick-filter-inline.component.ts:156]: It looks like overhead, why you can't use `this.submit()` method in place where you do `this.request$.next(this.value);`?

**Review by denisogolubev — COMMENTED** [review]
Run `lint` and `build:prod` locally to check that everything is working correctly.

File: packages/components/table/components/chips-filter-btn/chips-filter-btn.component.html [hunk 9]
```diff
@@ -0,0 +1,89 @@
+@switch (type) {
```
[the remark this answers is not in the document]
Comment (grankovsky) [reply packages/components/table/components/chips-filter-btn/chips-filter-btn.component.html:1]: There is different in logic of show elements and values

File: packages/components/table/components/chips-filter-btn/chips-filter-btn.component.html [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+  @case (InputType.Input) {
+    <div
+      (click)="clickAction()"
+      [matMenuTriggerFor]="menu"
+      [ngClass]="{filled:(value||valueText),disabled:disabled}"
```
[the remark this answers is not in the document]
Comment (grankovsky) [reply packages/components/table/components/chips-filter-btn/chips-filter-btn.component.html]: fixed

File: packages/components/table/components/chips-filter-btn/chips-filter-btn.component.html [hunk 11]
```diff
@@ -0,0 +1,89 @@
...
+      class="chips-btn">
+      <div class="quick--holder">
+        @if (customIcon) {
+          <mat-icon class="small-icon start-icon">{{ customIcon }}</mat-icon>
+        }
+        @if (!customIcon) {
+          <mat-icon class="small-icon start-icon" [ngClass]="{hidden:(!isChecked||(!value&&!valueText))}">check
```
[the remark this answers is not in the document]
Comment (grankovsky) [reply packages/components/table/components/chips-filter-btn/chips-filter-btn.component.html]: fixed

File: packages/components/table/components/chips-filter-btn/chips-filter-btn.component.html [hunk 12]
```diff
@@ -0,0 +1,89 @@
...
+          </mat-icon>
+        }
+        <span class="chips-text"><span>{{ title|translate }}</span>
```
[the remark this answers is not in the document]
Comment (grankovsky) [reply packages/components/table/components/chips-filter-btn/chips-filter-btn.component.html]: fixed

File: packages/components/table/components/chips-filter-btn/chips-filter-btn.component.html [hunk 13]
```diff
@@ -0,0 +1,89 @@
...
+        <span *ngIf="value||valueText" class="value-txt"
+              [style.max-width]="textWidth ? textWidth : ''">: {{ (valueText || value)|translate }}</span>
```
[the remark this answers is not in the document]
Comment (grankovsky) [reply packages/components/table/components/chips-filter-btn/chips-filter-btn.component.html]: fixed

File: packages/components/table/components/xm-table-quick-filter-controls.component.ts (hunk 5 above)
Comment (grankovsky) [reply packages/components/table/components/xm-table-quick-filter-controls.component.ts]: fixed

File: packages/components/table/components/xm-table-quick-filter-controls.component.ts (hunk 6 above)
Comment (grankovsky) [reply packages/components/table/components/xm-table-quick-filter-controls.component.ts]: removed

File: packages/components/table/components/xm-table-quick-filter-inline.component.ts (hunk 8 above)
Comment (grankovsky) [reply packages/components/table/components/xm-table-quick-filter-inline.component.ts:160]: same logic at regular filters / can be side effect without  distinctUntilChanged(),
            debounceTime(this.DELAY),

**Review by denisogolubev — APPROVED** [review]

**Review by LevBD — APPROVED** [review]

**Review by x3mlestat — APPROVED** [review]

**Pull request merged**